control-plane/agents/src/bin/core/controller/scheduling/volume_policy/simple.rs

```diff
@@ -35,45 +35,83 @@ impl ResourcePolicy<AddVolumeReplica> for SimplePolicy {
+
 impl SimplePolicy {
+    /// SimplePolicy free space SortCriteria.
+    fn free_space() -> SortCriteria {
+        SortCriteria::new(
+            Criteria::new("free_space", FREE_SPACE_WEIGHT),
+            ValueGrading::Higher,
+            |pool_item| pool_item.pool().free_space().into(),
+        )
+    }
+
+    /// SimplePolicy number of volume group replicas on pool SortCriteria.
+    fn vg_replica_count() -> SortCriteria {
+        SortCriteria::new(
+            Criteria::new("vg_replica_count", N_REPL_WEIGHT),
+            ValueGrading::Lower,
+            |pool_item| pool_item.vg_replica_count().into(),
+        )
+    }
+
+    /// SimplePolicy number of replicas on pool SortCriteria.
+    fn n_replicas() -> SortCriteria {
+        SortCriteria::new(
+            Criteria::new("n_replicas", N_REPL_WEIGHT),
+            ValueGrading::Lower,
+            |item| item.len().into(),
+        )
+    }
+
+    /// SimplePolicy over commitment SortCriteria.
+    fn over_commitment() -> SortCriteria {
+        SortCriteria::new(
+            Criteria::new("n_replicas", OVER_COMMIT_WEIGHT),
+            ValueGrading::Lower,
+            |item| item.pool().over_commitment().into(),
+        )
+    }
+
     /// Sort pools using weights between:
-    /// 1. number of replicas (N_REPL_WEIGHT %)
+    /// 1. number of replicas or number of replicas of a vg (N_REPL_WEIGHT %)
     /// 2. free space         (FREE_SPACE_WEIGHT %)
     /// 3. overcommitment     (OVER_COMMIT_WEIGHT %)
-    pub(crate) fn sort_by_weights(a: &PoolItem, b: &PoolItem) -> std::cmp::Ordering {
-        const N_REPL_WEIGHT: Ranged = Ranged::new_const(25);
-        const FREE_SPACE_WEIGHT: Ranged = Ranged::new_const(40);
-        const OVER_COMMIT_WEIGHT: Ranged = Ranged::new_const(35);
-
-        let n_replicas = Criteria::new("n_replicas", N_REPL_WEIGHT);
-        let free_space = Criteria::new("free_space", FREE_SPACE_WEIGHT);
-        let over_commit = Criteria::new("over_commit", OVER_COMMIT_WEIGHT);
-
-        let (score_a, score_b) = WeightedScore::dual_values()
-            .weigh(n_replicas, ValueGrading::Lower, a.len(), b.len())
-            .weigh(
-                free_space,
-                ValueGrading::Higher,
-                a.pool().free_space(),
-                b.pool().free_space(),
-            )
-            .weigh(
-                over_commit,
-                ValueGrading::Lower,
-                a.pool().over_commitment(),
-                b.pool().over_commitment(),
-            )
-            .score()
-            .unwrap();
-
-        score_b.cmp(&score_a)
+    pub(crate) fn sort_by_weights(
+        request: &GetSuitablePoolsContext,
+        a: &PoolItem,
+        b: &PoolItem,
+    ) -> std::cmp::Ordering {
+        // Add conditional criteria
+        let builder = if request.volume_group.is_some() {
+            if request.num_replicas > 1 {
+                SortBuilder::new().with_sort_criteria(SimplePolicy::vg_replica_count)
```

Comment:
hmm I think we should also somehow weigh in n_replicas here? Otherwise it will be completely disregarded?

Reply:
We want to disregard it right? Otherwise the best fit logic might get hampered? Or should we use different weights for both and include both? What do you think?

Reply:
As discussed have broken the total weight into two parts and assigned to `vg replica count` and `total count`.